src/views/offerPreferences/EditBuyPreferences.tsx

```diff
@@ -382,7 +383,14 @@ function Filters() {
 
 function PatchOfferButton() {
   const [preferences] = usePreferenceContext();
-  const { maxPremium } = preferences;
+
+  const { maxPremium } = useOfferPreferences(
+    (state) => ({
+      maxPremium: state.buyPremium,
+    }),
+    shallow,
+  );
```

Comment:
This should be the maxPremium of the offer by default (meaning what we had before, so no change required here)